src/group_theory/exponent.lean

```diff
@@ -91,57 +113,85 @@ end
 @[to_additive]
 lemma exponent_min (m : ℕ) (hpos : 0 < m) (hm : m < exponent G) : ∃ g : G, g ^ m ≠ 1 :=
 begin
-  by_contradiction,
-  push_neg at h,
-  have hcon : exponent G ≤ m := exponent_min' G m hpos h,
+  by_contra' h,
+  have hcon : exponent G ≤ m := exponent_min' m hpos h,
   linarith,
 end
 
 @[simp, to_additive]
 lemma exp_eq_one_of_subsingleton [subsingleton G] : exponent G = 1 :=
 begin
   apply le_antisymm,
-  { apply exponent_min' _ _ nat.one_pos,
+  { apply exponent_min' _ nat.one_pos,
     simp },
   { apply nat.succ_le_of_lt,
-    apply exponent_pos_of_exists _ 1 (nat.one_pos),
+    apply exponent_pos_of_exists 1 (nat.one_pos),
     simp },
 end
 
 @[to_additive add_order_dvd_exponent]
 lemma order_dvd_exponent (g : G) : (order_of g) ∣ exponent G :=
-order_of_dvd_of_pow_eq_one (pow_exponent_eq_one G g)
+order_of_dvd_of_pow_eq_one $ pow_exponent_eq_one g
 
-@[to_additive]
-lemma exponent_dvd_of_forall_pow_eq_one (n : ℕ) (hpos : 0 < n) (hG : ∀ g : G, g ^ n = 1) :
-  exponent G ∣ n :=
+variable (G)
+
+@[to_additive exponent_dvd_of_forall_nsmul_eq_zero]
+lemma exponent_dvd_of_forall_pow_eq_one (G) [monoid G] (n : ℕ) (hpos : 0 < n)
+  (hG : ∀ g : G, g ^ n = 1) : exponent G ∣ n :=
 begin
   apply nat.dvd_of_mod_eq_zero,
   by_contradiction h,
   have h₁ := nat.pos_of_ne_zero h,
-  have h₂ : n % exponent G < exponent G := nat.mod_lt _ (exponent_pos_of_exists _ n hpos hG),
+  have h₂ : n % exponent G < exponent G := nat.mod_lt _ (exponent_pos_of_exists n hpos hG),
   have h₃ : exponent G ≤ n % exponent G,
-  { apply exponent_min' _ _ h₁,
+  { apply exponent_min' _ h₁,
     simp_rw ←pow_eq_mod_exponent,
     exact hG },
   linarith,
 end
 
-variable [fintype G]
-
 @[to_additive lcm_add_order_of_dvd_exponent]
-lemma lcm_order_of_dvd_exponent : (finset.univ : finset G).lcm order_of ∣ exponent G :=
+lemma lcm_order_of_dvd_exponent [fintype G] : (finset.univ : finset G).lcm order_of ∣ exponent G :=
 begin
   apply finset.lcm_dvd,
   intros g hg,
-  exact order_dvd_exponent G g
+  exact order_dvd_exponent g
 end
 
+@[to_additive]
+lemma exists_max_prime_pow_dvd_exponent {p : ℕ} (hp : p.prime) :
```

Comment:
Should the name of the lemma include `order_of` in some way?

Reply:
Yeah, I'm really lost for ideas for a name here; I'll lost in the naming suggestions thread.